### PR TITLE
Remove casting to int on increment ID which can be prefixed by a string

### LINKED ISSUE #34451: [Issue] Remove casting to int on increment ID which can be prefixed by a string
This issue is automatically created based on existing pull request: magento/magento2#33369: Remove casting to int on increment ID which can be prefixed by a string

---------
### Description (*)
- Remove casting to int on increment ID which can be prefixed by a string

### Manual testing scenarios (*)
1. Configure a string prefix to invoice increment IDs
2. Submit an order
3. Create an invoice
4. Go to customer account -> orders -> invoice -> print invoice
5. The increment ID should be there, but is "0" for numbers prefixed with a string

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by sippsolutions**
### Description (*)
- Remove casting to int on increment ID which can be prefixed by a string

### Manual testing scenarios (*)
1. Configure a string prefix to invoice increment IDs
2. Submit an order
3. Create an invoice
4. Go to customer account -> orders -> invoice -> print invoice
5. The increment ID should be there, but is "0" for numbers prefixed with a string

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#34451: Remove casting to int on increment ID which can be prefixed by a string

### CODE REVIEW

**Comment by mrtuvn** [comment]
Thanks for contribute! Can you add some screenshot show problem?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @sippsolutions,
Could you add escaping for the string, and cover it with some type of automated test?

File: app/code/Magento/Sales/view/frontend/templates/order/print/invoice.phtml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 <?php foreach ($_invoices as $_invoice) : ?>
 <div class="order-details-items invoice">
     <div class="order-title">
-        <strong><?= $block->escapeHtml(__('Invoice #')) ?><?= (int) $_invoice->getIncrementId() ?></strong>
+        <strong><?= $block->escapeHtml(__('Invoice #')) ?><?= $_invoice->getIncrementId() ?></strong>
```
Comment (ihor-sviziev) [inline app/code/Magento/Sales/view/frontend/templates/order/print/invoice.phtml]: We still need to have escaping here.
```suggestion
        <strong><?= $block->escapeHtml(__('Invoice #')) ?><?= $block->escapeHtml($_invoice->getIncrementId()) ?></strong>
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Now the changes looks good to me. Could you cover your changes with some test?

**Comment by sippsolutions** [comment]
Hi @ihor-sviziev, as the only change is in a phtml-File, I don't think this is coverable with tests. I tested it in a live instance though.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you fix the failing integration tests?
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/33369/51783dfe9bd8db7201749b7c7fe0d1cd/Integration/console-error-logs.html
https://github.com/magento/magento2/blob/a190a755cfea7ab11abd6d89e113b0753a5b1237/dev/tests/integration/testsuite/Magento/Sales/Block/Order/PrintOrder/InvoiceTest.php#L115-L135

I think fixing the integration test will automatically add the test coverage for this PR

**Comment by ihor-sviziev** [comment]
@magento run Integration Tests, Static Tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9208](https://jira.corp.magento.com/browse/ENGCOM-9208) has been created to process this Pull Request

**Pull request merged**